modules/reactor/test/tests/meshgenerators/assembly_mesh_generator/assembly_only.i

```diff
@@ -20,17 +20,37 @@
     duct_halfpitch = '0.68'
     mesh_intervals = '1 1 1'
     quad_center_elements = false
-    region_ids='1 2 5; 11 12 15'
+    region_ids = '1 2 5; 11 12 15'
+  []
+  [pin1]
```

Comment:
@yjung-anl
I am trying to fix all the remaining tests in the reactor module that are failing. See https://civet.inl.gov/job/1338882/ for failure details. With the new error checks in PatternedMeshGenerator, we need to make sure that any MeshGenerators in AssemblyMeshGenerator/inputs have boundary sidesets named 'left', 'right', 'top', and 'bottom'. To do this, I renamed pin1 to pin1_a and ran it through a SideSetsFromNormalsGenerator to get the desired boundaries. However, the AssemblyMeshGenerator does not like this and gives the following error:

```
The reactor_params input must be a ReactorMeshParams type MeshGenerator
Please check that a valid definition and name of ReactorMeshParams has been provided.
```

Could you please look into a fix for this?

Reply:
This is because the mesh metadata generated by `PinMeshGenerator` are lost when you modify the mesh using SideSetsFromNormalsGenerator. @loganharbour @GiudGiud  and I are working on the mesh metadata forwarding capability development that will eventually solve this issue. I will help fix this once #23353 is merged.

Reply:
We were able to implement a fix in the source code that does not requires modification of the test input files.